ext/ManoptJuMPExt.jl

```diff
@@ -456,17 +485,17 @@ function JuMP.in_set_string(mime, set::ManifoldsBase.AbstractManifold)
 end
 
 """
-    JuMP.build_variable(::Function, func, m::ManifoldsBase.AbstractManifold)
+    JuMP.build_variable(::Function, func, manifold::ManifoldsBase.AbstractManifold)
 
-Build a `JuMP.VariablesConstrainedOnCreation` object containing variables
+Build a [`JuMP.VariablesConstrainedOnCreation`](@extref) object containing variables
 and the [`Manopt.JuMP_VectorizedManifold`](@ref) in which they should belong as well as the
 `shape` that can be used to go from the vectorized MOI representation to the
-shape of the manifold, that is, [`Manopt.JuMP_ArrayShape`](@ref).
+shape of the manifold, that is, [`Manopt.ManifoldDataShape`](@ref).
 """
-function JuMP.build_variable(::Function, func, m::ManifoldsBase.AbstractManifold)
-    shape = _point_shape(m)
+function JuMP.build_variable(::Function, func, manifold::ManifoldsBase.AbstractManifold)
```

Comment:
@blegat I have a question on this line, since I am currently fighting a bit with technical details in extensions: What is `func` here? A point on `manifold` in its original form? The `vectorize`(l. 498) seems to indicate so, but this is nowhere documented nor would the name make sense then.

Reply:
Good point, that's a bit tricky. Here, we let the user create the right shape. For instance here:
https://github.com/JuliaManifolds/Manopt.jl/blob/56ba724b31f0e13b5474906ab59b533ed4f4d821/test/MOI_wrapper.jl#L90
The user specifies that it's a 2x2 matrix so `func` is a 2x2 matrix.
If you need the variables to be of a shape that is not an array that won't work.
Then we'll have to go the route of PolyJuMP.jl were we have custom non-array shapes but that will use some JuMP dark magics as you like to call it so I'll need to make a PR :)

Reply:
I had a super small idea but looking at your explanation here again, I got lost in how this is supposed to ever work without `func` being an array, mainly because the magic you refer to in `PolyJuMP` also is 100% undocumented, so I have still no chance to understand how `func` is supposed to work nor which type it is. 

AllI could find is https://jump.dev/JuMP.jl/stable/api/JuMP/#JuMP.VariableInfo and that helps some new person like me with the feeling that my knowledge on JuMP is strictly upper bounded by zero – exactly no info how the “may be used” maybe can actually be used

Even though for now I just mainly look towards fixing the tutorial – I am by now also lost on that PR – so basically on all open PRs.